File: Python/problem052.py
#! /usr/bin/env python3
'''
Problem 52 - Project Euler
http://projecteuler.net/index.php?section=problems&id=052
'''
def samedigits(x,y):
    return set(str(x)) == set(str(y))

if __name__ == '__main__':
    maxn = 6
    x = 1
    while True:
        for n in range(2, maxn+1):
            if not samedigits(x, x * n):
                break
        else:
            print([x * i for i in range(1, maxn+1)])
            break
        
        x += 1

File: Kotlin/problem052.kt
// Problem 52 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=52
fun main(args: Array<String>) {
    var i = 100000
    while (!hasSameDigits(i)) i++
    println(i)
}

fun hasSameDigits(x: Int): Boolean {
    val dset = digitSet(x)
    if (dset.size < x.toString().length) {
        return false
    }
    for (i in 2..6) {
        if (dset != digitSet(x * i)) {
            return false
        }
    }
    return true
}

fun digitSet(x: Int): Set<Int> {
    return x.toString().map { it.code - 48 }.toSet()
}
